### PR TITLE
[PHPMD] Enable Sider's recommended set of rules

### PR DESCRIPTION
**Pull request by sleeeto**
> Explain a summary, purpose, or background for this change.

Enable the recommended rulset of PHPMD.

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

https://github.com/sider/sideci/issues/10670

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: images/phpmd/Dockerfile.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <%= render_erb 'images/Dockerfile.php.erb' %>
 
-COPY --chown=<%= chown %> images/<%= analyzer %>/sider_config.xml ${RUNNER_USER_HOME}/
+COPY --chown=<%= chown %> images/<%= analyzer %>/sider_recommended_phpmd.xml ${RUNNER_USER_HOME}/
```
Comment (ybiquitous) [inline images/phpmd/Dockerfile.erb:3]: [question] When do you remove `sider_config.xml`?

Comment (sleeeto) [reply images/phpmd/Dockerfile.erb:3]: I am going to remove all old config files at end of July.

Comment (ybiquitous) [reply images/phpmd/Dockerfile.erb:3]: OK, is there an issue? If not, please create an issue not to forget. 🙏🏼

**Review by ybiquitous — APPROVED** [review]
👍🏼

**Review by suetsuguk — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**